api/v1alpha1/etcdcluster_types.go

```diff
@@ -0,0 +1,47 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// EtcdClusterSpec defines the desired state of EtcdCluster
+type EtcdClusterSpec struct {
+	// Number of replicas in the cluster
+	// TODO: Investigate CRD scale sub-resources. See:
+	// https://book.kubebuilder.io/reference/generating-crd.html#scale
+	// +kubebuilder:validation:Minimum=0
+	Replicas int32 `json:"replicas"`
```

Comment:
This should be a `*int32` so that we can recognise no replicas specified and default it 😄

Also, do we want to support `replicas = 0`?

Reply:
I agree, but we actually copied it from the PoC.
I'll remove it since this has no bearing on the service creation stuff in this branch.

Reply:
Yeah just bin `replicas` until the next PR I think. :+1:

I actually don't think we should default it though. If you don't tell us how many replicas you want we can't really just make it up.